src/api/libcellml/analyserequationast.h

```diff
@@ -0,0 +1,354 @@
+/*
+Copyright libCellML Contributors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#pragma once
+
+#include "libcellml/analyser.h"
+#include "libcellml/generator.h"
+
+#ifdef NAN
+#    undef NAN
+#endif
+
+namespace libcellml {
+
+/**
+ * @brief The AnalyserEquationAst class.
+ *
+ * The AnalyserEquationAst class is for representing an equation abstract syntax
+ * tree (AST) in the context of a CellML Analyser.
+ */
+class LIBCELLML_EXPORT AnalyserEquationAst
+{
+    friend class Analyser;
+
+public:
+    /**
+     * @brief The type of a node in an abstract syntax tree (AST).
+     *
+     * A node in an abstract syntax tree (AST) can be of one of the following
+     * types:
+     *  - Assignment type:
+     *     - ASSIGNMENT: the assignment operator
+     *  - Relational and logical operators:
+     *    - EQ: the equal to function;
+     *    - NEQ: the not equal to function;
+     *    - LT: the less than function;
+     *    - LEQ: the less than or equal to function;
+     *    - GT: the greater than function;
+     *    - GEQ: the greater than or equal to function;
+     *    - AND: the and function;
+     *    - OR: the or function
+     *    - XOR: the exclusive or function;
+     *    - NOT: the not function;
+     *  - Arithmetic operators:
+     *    - PLUS: the plus operator;
+     *    - MINUS: the minus operator;
+     *    - TIMES: the times operator;
+     *    - DIVIDE: the divide operator;
+     *    - POWER: the power operator;
+     *    - ROOT: the root operator;
+     *    - ABS: the absolute value function;
+     *    - EXP: the exponential function;
+     *    - LN: the Napierian logarithm function;
+     *    - LOG: the common logarithm function;
+     *    - CEILING: the ceiling function;
+     *    - FLOOR: the floor function;
+     *    - MIN: the minimum function;
+     *    - MAX: the maximum function;
+     *    - REM: the remainder function;
+     *  - Calculus elements:
+     *    - DIFF: the differentiation operator;
+     *  - Trigonometric operators:
+     *    - SIN: the sinus function;
```

Comment:
should be "sine" and "cosine" in English

Reply:
also below with "arc" and "hyperbolic"

Reply:
Done in commit 0f277e0a.